maps-compose-utils/src/main/java/com/google/maps/android/compose/clustering/Clustering.kt

```diff
@@ -0,0 +1,113 @@
+package com.google.maps.android.compose.clustering
+
+import android.os.Handler
+import android.os.Looper
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.platform.LocalContext
+import com.google.android.gms.maps.GoogleMap
+import com.google.maps.android.clustering.Cluster
+import com.google.maps.android.clustering.ClusterItem
+import com.google.maps.android.clustering.ClusterManager
+import com.google.maps.android.collections.MarkerManager
+import com.google.maps.android.compose.CameraPositionState
+import com.google.maps.android.compose.InputHandler
+import com.google.maps.android.compose.MapEffect
+import com.google.maps.android.compose.MapsComposeExperimentalApi
+import com.google.maps.android.compose.rememberReattachClickListenersHandle
+
+/**
+ * Groups many items on a map based on zoom level.
+ *
+ * @param items all items to show
+ * @param cameraPositionState the state of the camera, which affects clustering behavior
+ * @param onClusterClick a lambda invoked when the user clicks a cluster of items
+ * @param onClusterItemClick a lambda invoked when the user clicks a non-clustered item
+ * @param onClusterItemInfoWindowClick a lambda invoked when the user clicks the info window of a
+ * non-clustered item
+ * @param onClusterItemInfoWindowLongClick a lambda invoked when the user long-clicks the info
+ * window of a non-clustered item
+ */
+@MapsComposeExperimentalApi
+@Composable
+public fun <T : ClusterItem> Clustering(
+    items: Collection<T>,
+    cameraPositionState: CameraPositionState,
```

Comment:
Suggestion/idea: how about making this a part of the `MapApplier` so that you can access it implicitly?

Reply:
Was wondering if that'd be prudent. I think I'll go for it.

Reply:
Leaning towards a non-providable local, to avoid giving MapApplier even more responsibilities.

Reply:
Actually, a public Composable property seems better, since then I can annotate it with `@GoogleMapComposable`.